### DISCUSSION TITLE
Error in MCP connection to server sqlite: Error: spawn uvx ENOENT

### DISCUSSION THREAD

**Discussion by kun321** [modelcontextprotocol/.github#20] [Q&A] [answered]
When I tried quickStart, I created the sqlite3 database and configured claude config.jso. After restarting, I got some errors
``` log
2024-11-27T02:39:10.944Z [info] Attempting to connect to MCP server sqlite...
2024-11-27T02:39:10.980Z [error] Error in MCP connection to server sqlite: Error: spawn uvx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn uvx',
  path: 'uvx',
  spawnargs: [Array]
}
2024-11-27T02:39:10.980Z [error] Could not start MCP server sqlite: Error: spawn uvx ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn uvx',
  path: 'uvx',
  spawnargs: [Array]
}
```

and i make some try follow the [Troubleshooting](https://modelcontextprotocol.io/quickstart#mcp-or-database-errors)
![image](https://github.com/user-attachments/assets/149d5baf-9951-47cd-8745-09f69e47392d)

**Comment by kun321** [comment, marked as answer]
I have solved the problem. 
Uv installation with brew
MUST USE `brew install uv`

**Reply by CindyXing** [reply]
Thanks. This helped me.

**Reply by andresgaona29** [reply]
I've do this with pip install and it works too

**Reply by crock** [reply]
Yep, this worked for me! thanks

**Reply by PunchBui** [reply]
This work for me! thanks!

**Comment by S1M0N38** [comment]
I encountered the same issue while following the Server tutorial in the docs. I had installed uv using `pipx`. I solved the issue by providing the full path to the uv command (in my case `/User/simo/.local/bin/uv`).

I think the Claude Desktop App does not source the file where I added `/User/simo/.local/bin` to the PATH, and for this reason, it wasn't able to find the uv command (despite having uv installed).

**Reply by mmilbourne** [reply]
I had the same problems on Windows.  uv.exe was installed in the <USER>\.local\bin directory.  Specifying the full path fixed the problem.

**Reply by shaofengshi** [reply]
Solved by your suggestion. Thanks!

**Comment by camball** [comment]
Took a look and didn't like how the uv/uvx installer did it.

They added a file `~/.local/bin/env` which is a shell script that was run from my shell startup script (`~/.zshrc` in my case) and added everything in `~/.local/bin/` to my path. No idea why it's not just a simple "add folder to $PATH", so that's what I changed it to. I deleted the `env` and `env.fish` files to not leave stale scripts. I also moved the initialization from `~/.zshrc` to `~/.zprofile` so that it runs on non-interactive shells as well as interactive shells, as I was thinking maybe the shell that is running the MCP connection (inside Cursor, in my case) is non-interactive.

To `~/.zprofile`:

```sh
# uv / uvx
export PATH="$HOME/.local/bin:${PATH}"
```

When I made these changes, worked great and MCP server started up just fine. I didn't want to have to install with Homebrew.

```json
{
  "mcpServers": {
    "awslabs.cfn-mcp-server": {
      "command": "uvx",
      "args": ["awslabs.cfn-mcp-server@latest", "--readonly"],
      "autoApprove": []
    }
  }
}

```

**Comment by davutg** [comment]
on windows installing by below and adding this to ENV PATH vars solved C:\Users\[YourUserName]\.local\bin 

`powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"`

**Comment by mkalfs** [comment]
# Fix: MCP Windows-MCP spawn uv ENOENT (Claude Desktop / Cowork)

## Problem

When using Claude Desktop with Cowork or MCP servers, you may encounter the following errors:

```
MCP Windows-MCP: spawn uv ENOENT
MCP Windows-MCP: Server disconnected
```

In some cases, you may also see:

```
Reinstall required (Cowork requires Claude Desktop…)
```

## Root Cause

The error `spawn uv ENOENT` means that the system attempted to execute the `uv` command, but Windows could not locate it.

This typically occurs when:

* `uv` is not installed
* `uv` is installed but not available in the system PATH
* Claude Desktop cannot access the PATH environment

## Step 1 — Verify Installation

Open PowerShell and run:

```
uv --version
where.exe uv
```

If you receive an error such as:

```
The term 'uv' is not recognized...
```

then `uv` is not installed.

## Step 2 — Install uv

If `winget` is available:

```
winget install --id=astral-sh.uv -e
```

If `winget` is not available, use the official installer:

```
powershell -ExecutionPolicy Bypass -c "irm https://astral.sh/uv/install.ps1 | iex"
```

## Step 3 — Restart Shell

Close PowerShell completely and open a new window. Then verify again:

```
uv --version
where.exe uv
```

You should now see a version and a path to `uv.exe`.

## Step 4 — If uv Is Still Not Found

Check common install locations:

```
Test-Path "$env:USERPROFILE\.local\bin\uv.exe"
Test-Path "$env:USERPROFILE\AppData\Roaming\uv\bin\uv.exe"
```

If one returns `True`, test directly:

```
& "$env:USERPROFILE\.local\bin\uv.exe" --version
```

## Step 5 — Add uv to PATH Manually

If the executable works but is not globally available, add it to PATH:

```
[Environment]::SetEnvironmentVariable(
  "Path",
  $env:Path + ";$env:USERPROFILE\.local\bin",
  "User"
)
```

Then restart PowerShell and verify again.

## Step 6 — Update MCP Configuration

Even if `uv` works in PowerShell, Claude Desktop may still not detect it.

It is recommended to use the absolute path to `uv.exe` in your MCP configuration.

Example:

```
{
  "mcpServers": {
    "windows-mcp": {
      "command": "C:\\Users\\YourUser\\.local\\bin\\uv.exe",
      "args": [
        "run",
        "windows-mcp"
      ]
    }
  }
}
```

Replace the path with the one returned by:

```
where.exe uv
```

## Step 7 — Reboot and start Claude Desktop

After installation and configuration:

* Reboot Windows
* Reopen the application


## Notes

* The "Server disconnected" message is a secondary error caused by the failure to start `uv`
* The "Reinstall required" message may persist until Claude Desktop is restarted or reinstalled

## Summary

If you encounter the error:

```
spawn uv ENOENT
```

it means that `uv` is not installed or not accessible.

Installing `uv` and ensuring it is correctly referenced in the MCP configuration resolves the issue.

## Checklist

* uv is installed
* uv is accessible via PATH or full path is used
* MCP config points to correct executable
* Claude Desktop has been restarted
